src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala

```diff
@@ -107,6 +121,10 @@ object EasyArchiveBag extends Bagit5FacadeComponent with DebugEnhancedLogging {
     }
   }
 
+  private def createFailedHttpRequestException(bagId: BagId, get: HttpGet, statusLine: StatusLine): IllegalStateException = {
+    new IllegalStateException(s"Error retrieving bag-sequence for bag: $bagId. [${ get.getURI }] returned ${ statusLine.getStatusCode } ${ statusLine.getReasonPhrase }")
```

Comment:
this error message looks very much like the log message above. A couple points for your consideration:
* would it be appropriate to add similar logging to the `case _ => throw createFailedHttpRequestException(bagId, get, statusLine)` above? If so, you should add the `logger.error(...)` line above to this function as well.
* where does this `IllegalStateException` end up? My guess would be that it travels all the way up to `easy-ingest-flow`, where it will be caught and put into `deposit.properties` as the `state.description`. This is exposed to external users via both SWORD2 and deposit-ui. Hence, I think the error message should contain less technical information. The current message should be the one that is logged, containing all the detailed technical information, but the exception itself should IMHO contain less.